tests.js

```diff
@@ -9,14 +9,51 @@
     return [template, values];
   }
 
+  function r(template, ...values) {
+    if (values.length !== 0) {
+      throw new Error();
+    }
+    return template.raw[0];
+  }
+
   const tests = [
+    // No interpolations
+    [...qpair`^foo\(bar\);\n$`, '',
+     r`/^foo\(bar\);\n$/`],
+    // No interpolations but flags
+    [...qpair`^foo\(bar\);\n$`, 'gi',
+     r`/^foo\(bar\);\n$/gi`],
+    // A single string into a block context.
     [...qpair`^${ 'foo' }$`, '',
-     '/^(?:foo)$/'],
+     r`/^(?:foo)$/`],
+    // Testing transitions between contexts.
     [...qpair`^([${ '\\' }${ /[a-z]/ }]{${ 42 }})${ /$/ }`, 'i',
-     '/^([\\\\a-z]{42})(?:$)/i'],
-    // See above for why this says "\$${" rather than "\${"
+     r`/^([\\a-z]{42})(?:$)/i`],
+    // Back-reference not scoped to containing RegExp
     [...qpair`^(#+)([^#\r\n]*)${ /\1/ }`, '',
      '/^(#+)([^#\\r\\n]*)(?:\\1)/'],
+    // Negated charset into a charset
+    [...qpair`[${ /[^A-Z]/ }]`, '',
+     r`/[\u0000-@\[-\uffff]/`],
+    // String into a charset
+    [...qpair`[${ "A-Z" }]`, '',
+     r`/[A\-Z]/`],
+    // String into a negated charset
+    [...qpair`[^${ "A-Z" }]`, '',
+     r`/[^A\-Z]/`],
+    // Multiple elements into a charset: individual chars, charsets,
+    // and special groups.
+    [...qpair`[${ /[a]|([c]|b)|d|_/ }]`, '',
+     r`/[_a-d]/`],
+    // Multiple case-insensitive elements into a charset: individual chars,
+    // charsets, and special groups.
+    [...qpair`[${ /[a]|(?:[c]|b)|d|_/i }]`, '',
+     r`/[A-D_a-d]/`],
+    [...qpair`[${ /x{1,2}/ }]`, '',
+     r`/[x]/`],
+    // \b means different things.
```

Comment:
Does this \b issue relate to a nearby line of code? If not, did you mean it to be a TODO?